src/transactions/test/SorobanTxTestUtils.cpp

```diff
@@ -545,9 +548,69 @@ ContractInvocationTest::computeResourceFee(SorobanResources const& resources,
     return sorobanResourceFee(*mApp, resources, txSize, 0);
 }
 
+WasmContractInvocationTest::WasmContractInvocationTest(
+    RustBuf const& wasm, bool deployContract, Config cfg, bool useTestLimits,
+    std::function<void(SorobanNetworkConfig&)> cfgModifyFn)
+    : ContractInvocationTest(cfg, useTestLimits, cfgModifyFn), mWasm(wasm)
+{
+    if (deployContract)
+    {
+        // Deploy with default resources
+        SorobanResources uploadResources{};
+        uploadResources.instructions = 200'000 + (mWasm.data.size() * 6000);
+        uploadResources.readBytes = 1000;
+        uploadResources.writeBytes = 5000;
+
+        SorobanResources createResources{};
+        createResources.instructions = 200'000;
+        createResources.readBytes = 5000;
+        createResources.writeBytes = 5000;
+
+        deployWithResources(uploadResources, createResources);
+    }
+}
+
+void
+WasmContractInvocationTest::deployWithResources(
+    SorobanResources const& uploadResources,
+    SorobanResources const& createResources)
+{
+    if (!mContractKeys.empty())
+    {
+        throw "WASM already uploaded";
```

Comment:
nit: Wasm

Also, why does this throw? It's safe to re-upload Wasm.

Reply:
This is just meant to be an initializer function, so there's no need to call it more than once.

Reply:
Ok. Still, please update WASM->Wasm

Reply:
Yep that's been updated